### PR TITLE
Delete old duplicate matching UI code

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

Removing the old duplicate matching UI code in preperation for when we switch on the feature flag for the new code. The actual feature flag will be removed in a separate PR with the data migration.

## Changes proposed in this pull request

 Delete old duplicate matching UI code

- Delete UI
- Delete controller
- Delete components
- Delete routes
- Replace path with the new routes
- Delete fraud_match email 

## Guidance to review

- Is all the code removed?
- Did I miss anything? 
- Have I removed anything extra?

## Link to Trello card

https://trello.com/c/jUjJW2kp/4251-delete-old-duplicate-matching-ui-code

### CODE REVIEW

**Review by stevehook — COMMENTED** [review]
This PR leaves the feature flag in place but makes it redundant. Would it make sense to just drop the feature flag in this PR too (with a follow up PR just to do the data migration)? Otherwise somebody might think that they can revert back to the old behaviour by deactivating the feature flag in the support interface?

**Comment by tomas-stefano** [comment]
Yes, I agree, although if we remove the feature flag I'd prefer to do the data migration in the same PR (could it be everything in this PR or another PR specifically for the removal of the feature flag + data migration).

**Comment by tomas-stefano** [comment]
@stevehook I removed the duplicate matching feature flag on this PR along with the data migration. Can you review it again, please?

File: lib/tasks/data.rake [hunk 1]
```diff
@@ -1,5 +1,6 @@
 DATA_MIGRATION_SERVICES = [
   # do not delete or edit this line - services added below by generator
+  'DataMigrations::RemoveDuplicateMatchingFeatureFlag',
```
[the remark this answers is not in the document]
Comment (tomas-stefano) [reply lib/tasks/data.rake:3]: I see, so the PR recommendation "If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database" is good but can have some drawbacks as you mention. It is not an issue thought in this case. Thank you for raising 🙏

**Pull request merged**